publisher.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"fmt"
+	"time"
+	"net/http"
+	"io/ioutil"
+	"log"
+	"github.com/jlaffaye/ftp"
+	"bytes"
+	"sync"
+	"io"
+)
+
+var lock = &sync.Mutex{}
+var wg = &sync.WaitGroup{}
+
+
+func processFiles(files <- chan string) {
+	defer wg.Done()
+	conn, err := connectToFtp()
+	if err != nil {
+		log.Fatal("Unable to connect to FTP server exiting - aborting")
+	}
+	defer conn.Logout()
+	defer conn.Quit()
+	for file := range files {
+		buf, err := getFileFromFTP(file, conn)
+		if err == nil {
+			status, err := postFileToRas(file, &buf)
+			if status == 200 && err == nil {
+				deleteFileOnFTP(conn, file)
+			}
+		}
+	}
+
+}
+func deleteFileOnFTP(conn *ftp.ServerConn, file string) {
+	err := conn.Delete(file)
+	if err != nil {
```

Comment:
this could be inline with the if statement: < if err := conn.Delete(file); err !=nil { >